include/rocksdb/env.h

```diff
@@ -163,6 +164,33 @@ class Env {
   // The result of Default() belongs to rocksdb and must never be deleted.
   static Env* Default();
 
+  // Handles the event when a new DB or a new ColumnFamily starts using the
+  // specified data paths.
+  //
+  // The data paths might be shared by different DBs or ColumnFamilies,
+  // so OnDbPathsRegistered might be called with the same data paths.
+  // For example, when CreateColumnFamily is called multiple times with the same
+  // data path, OnDbPathsRegistered will also be called with the same data path.
+  //
+  // When a path is called in OnDbPathsRegistered, it must be correspondingly
+  // called in OnDbPathsUnregistered.
+  //
+  // Different implementations may take different actions.
+  // By default, it's a no-op.
+  virtual Status OnDbPathsRegistered(const std::set<std::string>& /*paths*/) {
```

Comment:
agree, maybe in the future, there are certain handlers that assume the paths can be duplicated.

Reply:
According to the method's doc:  `Handles the event when a new DB or a new ColumnFamily starts using the specified data paths.` Using a `set` here enforces that the data paths should not be duplicated in the parameter.

Reply:
@siying  If it's a more general API like "OnPathsRegistered", then I think it's more natural for the paths to be a `vector`. I'm ok with making this API more general, what do you think?

Reply:
After second thought, I'm ok with making it a `vector`, then also in `DBImpl`, we do not deduplicate paths via `set`, we always respect what users have set in `dbname`, `db_paths` and `cf_paths`.

Reply:
It makes sense. You can decide.

Reply:
Considering a case: Open a db with specified 2 `ColumnFamilyDescriptor`s which share a `cf_path` , then if using a `set` of paths, only 1 path and 1 reference to that path is added to the cache. Later, `DropColumnFamily` on one of the column, then the cached path will be evicted since ref count reaches 0, but there is still another column using the path, so the cached size should be kept there.
Using a `vector` of paths and do not deduplicate them can solve this problem.

Reply:
I didn't suggest that DBImpl should pass duplicated value with std::vector<>. We can dedup in DBImpl and pass a std::vector<>. But you have a point that, as a general API, users of the API could pass duplicated value and we need to handle it. So it's your call.